corehq/apps/userreports/tests/test_save_errors.py

```diff
@@ -61,3 +66,32 @@ def test_missing_column(self):
         }
         with self.assertRaises(MissingColumnWarning):
             adapter.best_effort_save(doc)
+
+
+class AdapterBulkSaveTest(TestCase):
+
+    def setUp(self):
+        self.domain = 'adapter_bulk_save'
+        self.config = get_sample_config(domain=self.domain)
+        self.config.save()
+        self.adapter = get_indicator_adapter(self.config, raise_errors=True)
+
+    def tearDown(self):
+        self.config.delete()
+        self.adapter.clear_table()
+
+    def test_bulk_save(self):
+        docs = []
+        for i in range(10):
+            docs.append({
+                "_id": str(i),
+                "domain": self.domain,
+                "doc_type": "CommCareCase",
+                "name": 'doc_name_' + str(i)
+            })
+
+        self.adapter.build_table()
+        self.adapter.bulk_save(docs)
```

Comment:
Sure

Reply:
Unfortunately since the queries are executed in sqlalchemy, Django's `assertNumQueries` doesn't capture the query count. sqlalchemy does provide a way to count the queries using its event listener framework, but it needs the sqlalchemy cursor object. So it's not possible to do that here without actually modifying `bulk_save` itself.